### ISSUE TITLE
How do you run this in some kind of daemon mode?

### ISSUE THREAD

**Issue by flavorgold1** [AdguardTeam/dnsproxy#232]
I'm on Ubuntu Linux x64.

**Comment by joshcangit** [comment]
Check out my repo https://github.com/joshcangit/dnsproxy-service for setting up a systemd service.

Not using `dnsmasq` but I'm using `systemd-resolved`.
First, I set this in  `/etc/resolvconf/resolv.conf.d/tail`.
```
nameserver 127.0.0.1
nameserver ::1
```
Then, I run `sudo resolvconf -u`.

After that, I set this in `/etc/systemd/resolved.conf`.
```
[Resolve]
#DNS=
#FallbackDNS=
Domains=~.
#LLMNR=no
MulticastDNS=yes
DNSSEC=yes
DNSOverTLS=yes
#Cache=no-negative
DNSStubListener=no
#ReadEtcHosts=yes
```
Finally, I run `sudo systemctl restart systemd-resolved`.
I didn't even needed to remove `systemd-resolved` or set a different port.

**Comment by joshcangit** [comment]
It's not possible. It needs to be able to build the DNS cache on its own.
I've only managed to make it run after the DNS cache is built by _systemd-resolved_.
However after boot up, DNS will never resolve because `dnsproxy` needs to have cache already available.

**Comment by Lanius-collaris** [comment]
> It's not possible. It needs to be able to build the DNS cache on its own. I've only managed to make it run after the DNS cache is built by _systemd-resolved_. However after boot up, DNS will never resolve because `dnsproxy` needs to have cache already available.

The reason may be that `dnsproxy` can't get the IP of upstream. `-b` can specify bootstrap DNS, for example: `-b 45.90.28.1:5353`. You can also use DNS Stamp or some servers which have SSL certificates for IP address. 
`dnsproxy` has cache, use `--cache` to enable it.

**Comment by joshcangit** [comment]
> The reason may be that `dnsproxy` can't get the IP of upstream. `-b` can specify bootstrap DNS, for example: `-b 45.90.28.1:5353`. You can also use DNS Stamp or some servers which have SSL certificates for IP address. `dnsproxy` has cache, use `--cache` to enable it.

Nope. I have set `bootstrap` addresses and enabled `cache`. DNS still fail after system boot up.

**Comment by joshcangit** [comment]
> The reason may be that `dnsproxy` can't get the IP of upstream. `-b` can specify bootstrap DNS, for example: `-b 45.90.28.1:5353`. You can also use DNS Stamp or some servers which have SSL certificates for IP address. 
> `dnsproxy` has cache, use `--cache` to enable it.

No, I do use `bootstrap` addresses and enable `cache` for `dnsproxy`

Just needs to modiy `/etc/resolv.conf` for any listen address `dnsproxy` uses.
https://github.com/AdguardTeam/dnsproxy/issues/275#issuecomment-1255806799

**Comment by joshcangit** [comment]
> I'm on Ubuntu Linux x64.

Same here. I have now publish a repository just for this.
https://github.com/joshcangit/dnsproxy-systemd
